codecs/avif/Makefile

```diff
@@ -106,8 +110,29 @@ $(LIBWEBP_DIR)/CMakeLists.txt: $(CODEC_DIR)/CMakeLists.txt
 	curl -sL $(LIBWEBP_URL_WITH_SHARPYUV) \
 		| tar xzm -C $(LIBWEBP_DIR)
 
-# Make libsharpyuv.a
-$(LIBSHARPYUV): $(LIBWEBP_DIR)/CMakeLists.txt
+# Make libsharpyuv.a for ST-Encoding
+$(LIBSHARPYUV_ST): $(LIBWEBP_DIR)/CMakeLists.txt
+	mkdir -p $(@D)
+	emcmake cmake \
+		-DCMAKE_DISABLE_FIND_PACKAGE_Threads=1 \
```

Comment:
I tried using the suggested `-DWEBP_USE_THREAD=OFF`, however, it still resulted in the same error seen in #1397

Any ideas @wantehchang? 🤔

Reply:
@wantehchang Any downsides in using the flag used here instead of the one you recommended? I had a quick sift through the build files and I suspect that flag isn’t forwarded correctly to the yuv build, but not 100% sure.

Reply:
Jamie, Surma: I am sorry I forgot to reply. This seems like a bug in `-DWEBP_USE_THREAD=OFF`. I am not familiar with the build system of libwebp (of which libsharpyuv is a component). @jzern James, is `-DWEBP_USE_THREAD=OFF` the right cmake option to disable all threading code (including the dependency on the pthread library) in libwebp?

Reply:
```
$ emcmake cmake .. -DWEBP_BUILD_WEBP_JS=ON
$ emmake make -j
$ grep WEBP_USE_THREAD src/webp/config.h 
/* #undef WEBP_USE_THREAD */
```

So maybe I'm missing something with the build setup in Squoosh.

Reply:
> So maybe I'm missing something with the build setup in Squoosh.

I wasn't looking at the diff.

```
$ emcmake cmake .. \
  -DWEBP_BUILD_ANIM_UTILS=OFF \
  -DWEBP_BUILD_CWEBP=OFF \
  -DWEBP_BUILD_DWEBP=OFF \
  -DWEBP_BUILD_GIF2WEBP=OFF \
  -DWEBP_BUILD_IMG2WEBP=OFF \
  -DWEBP_BUILD_VWEBP=OFF \
  -DWEBP_BUILD_WEBPINFO=OFF \
  -DWEBP_BUILD_LIBWEBPMUX=OFF \
  -DWEBP_BUILD_WEBPMUX=OFF \
  -DWEBP_BUILD_EXTRAS=OFF \
  -DBUILD_SHARED_LIBS=OFF \
  -DCMAKE_BUILD_TYPE=Release
...
-- Found Threads: TRUE  
...
$ grep WEBP_USE_THREAD src/webp/config.h
#define WEBP_USE_THREAD 1
```

That's expected.

```
$ emcmake cmake .. \
  -DWEBP_USE_THREAD=OFF \
  -DWEBP_BUILD_ANIM_UTILS=OFF \
  -DWEBP_BUILD_CWEBP=OFF \
  -DWEBP_BUILD_DWEBP=OFF \
  -DWEBP_BUILD_GIF2WEBP=OFF \
  -DWEBP_BUILD_IMG2WEBP=OFF \
  -DWEBP_BUILD_VWEBP=OFF \
  -DWEBP_BUILD_WEBPINFO=OFF \
  -DWEBP_BUILD_LIBWEBPMUX=OFF \
  -DWEBP_BUILD_WEBPMUX=OFF \
  -DWEBP_BUILD_EXTRAS=OFF \
  -DBUILD_SHARED_LIBS=OFF \
  -DCMAKE_BUILD_TYPE=Release
...
# no mention of threads
...
$ grep WEBP_USE_THREAD src/webp/config.h 
/* #undef WEBP_USE_THREAD */
```

Reply:
Thanks. From what I can gather would it be best to use the flag `-DWEBP_BUILD_WEBP_JS=ON`? That seems to disable the use of the mutex, among other things.